include/rcutils/error_handling.h

```diff
@@ -103,6 +103,22 @@ rcutils_set_error_state(
 #define RCUTILS_SET_ERROR_MSG(msg, allocator) \
   rcutils_set_error_state(msg, __FILE__, __LINE__, allocator);
 
+/// Set the formatted error msg and free its memory allocated
+/**
+ * This function sets the error msg with the format specified and free the 
```

Comment:
This implies, at least to me, that a format string is given and extra arguments are expected to generate a string, a la `printf`, but that's no the case here. I think you mean something like "allocated string" or "non-literal" string or "string that should be deallocated after use".